parmed/openmm/parameters.py

```diff
@@ -101,10 +107,20 @@ def from_parameterset(cls, params, copy=False):
         new_params.parametersets = params.parametersets
         new_params._combining_rule = params.combining_rule
         new_params.residues = params.residues
+        # Assign Lennard-Jones atom classes
+        ljtypes = dict()
+        counter = 0
+        for name, atom_type in iteritems(new_params.atom_types):
+            key = (atom_type.rmin, atom_type.epsilon, atom_type.rmin_14,
+                    atom_type.epsilon_14)
+            if key not in ljtypes:
+                ljtypes[key] = counter
```

Comment:
Our primary concern is to translate the original parameter specification files (`*.dat` files for AMBER) into faithful representations of these parameter sets in `ffxml` files, ideally in a way that maximizes our ability to mix-and-match sets of these files.

I think there's value in ending up with a `<NonbondedForce>` block that essentially encodes the same information at the same level of detail as the [`parm96.dat` file does](https://github.com/choderalab/ambermini/blob/master/share/amber/dat/leap/parm/parm96.dat#L450-L488).

It sounds like your current approach tries to compress even further than what is specified by these type maps:
```
N   NA  N2  N*  NC  NB  NP  NO
C   C*  CA  CB  CC  CN  CM  CK  CQ  CW  CV  CR  CA  CX  CY  CD
```
but using numeric types---is that correct?

What if we compromised, and used a sensible reference atom type name as the class name?  I think this will be much more comprehensible to anyone looking at these files in the future than class numbers.